### ISSUE TITLE
String should not contain control characters.

### ISSUE THREAD

**Issue by ExtReMLapin** [IntrinsicLabsAI/gbnfgen#31]
Right now, strings can for example contain a new line, not an escaped one.

Which will lead to incorrect json generation

Here is my suggestion : 

```
string ::= "\"" (char | escapeSequence)* "\""
char ::= [^"\\\x7F\x00-\x1F]
escapeSequence ::= "\\\"" | "\\\\" | "\\n" | "\\r" | "\\t" | "\\b" | "\\f" 


```

I could open a PR but I could be missing something so I open an issue to discuss this

**Comment by ExtReMLapin** [comment]
I met this issue because on rare cases (15% of generations) i had this kind of json generated : 

```json
{"entities": [

{"type": "Geo-Political Entity", "name": "Paris 2024"},
{"type": "Location", "name": "Paris"}],
"relations": [{"type_from": "G',
eographic Feature", "name_from": "Paris", "type_to": "Event", "name_to": "Paris 2024", "relation_type": "occurs_in"}]}
```

**Comment by ExtReMLapin** [comment]
Entering manually \r and \n isn't enough, I ended up with having another control character getting outputed as JSON, which made my json un-usable

**Comment by ExtReMLapin** [comment]
Updated it to exclude singlequote

**Comment by a10y** [comment]
Fixed in #37

**Comment by ExtReMLapin** [comment]
🖖🏻
